conda_forge_tick/update_recipe/build_number.py

```diff
@@ -51,3 +58,28 @@ def update_build_number(raw_meta_yaml, new_build_number, build_patterns=None):
         raw_meta_yaml = "\n".join(lines) + "\n"
 
     return raw_meta_yaml
+
+
+def update_build_number_recipe_yaml(
+    raw_recipe_yaml: str, new_build_number: Callable[[str], str] | str
+):
+    def replace_build_number(recipe, first_key, second_key):
+        if first := recipe.get(first_key):
+            if second_key in first and isinstance(first[second_key], int):
+                if callable(new_build_number):
+                    first[second_key] = new_build_number(first[second_key])
+                else:
+                    first[second_key] = new_build_number
+
+    recipe = yaml.safe_load(raw_recipe_yaml)
+
+    cases = [
+        ("build", "number"),
+        ("context", "build_number"),
+        ("context", "build"),
+    ]
+
+    for case in cases:
+        replace_build_number(recipe, *case)
+
+    return yaml.dump(recipe, sort_keys=False)
```

Comment:
`yaml` sorts the keys per default when deserializing. I don't think we ever want that, so let's remember to set the argument also for other cases of `yaml.dump` in this and other PRs.